### DISCUSSION TITLE
Self-describing Dataset - EKG/Principles

### DISCUSSION THREAD

**Discussion by giscus** [EKGF/ekg-principles#21] [Page Comments]
# Self-describing Dataset - EKG/Principles

Manifesto and principles for the Enterprise Knowledge Graph (EKG)

https://principles.ekgf.org/vocab/sdd/

**Comment by DianeKEPR** [comment]
Would like to know better the definition/meaning of the term "Self-describing".    If it means that the dataset is defined intuitively by structure of the KG it was pulled from, or inferred from what might be a list of metadata about the dataset above, then in my view a narrative that describes the intended purpose and appropriate use of the dataset should also be available.   I do believe it is helpful to have the main use cases provided for the dataset, and also the use cases that the dataset is not appropriate for.   Business people are generally the primary user of pulled datasets and there is increasing regulatory scrutiny that the business must have knowledge about the intended use of the dataset and its data, if used in any decision making.

**Reply by jgeluk** [reply]
An SDD is a "logical dataset", although the distinction between logical and physical is fuzzy in an EKG and often quite the same. I'm not sure if I follow your second sentence. An EKG is a collection of these logical datasets. They can be physically distributed across many different places, triplestores or even other types of databases and one given SDD can have multiple physical manifestations in multiple locations of the overall distributed EKG.

About the use cases: of course people would like to know which use cases are available for a given dataset yes but that information does not necessarily have to be stored in that dataset itself which is what I meant with "use cases point to the dataset and not vice versa". You could still query the EKG for "give me all use cases for dataset X". There is a little bit of a paradox going on there: on one hand you do not want to design an SDD for any particular use case (because that would make it biased and less reusable) and on the other hand you want to provide focus in the development of that SDD by designing if for your first use cases :-). There's a balance to find there which is one reason to have a proper separation of concerns here: everything that's use-case-specific goes into the use case data (which is its own dataset) and what remains in the SDD is "generic".
